samples/server/go-server/api/server/v1alpha1/impl.go

```diff
@@ -43,9 +44,19 @@ func (g *Grafeas) CreateNote(ctx context.Context, req *pb.CreateNoteRequest) (*p
 		return nil, status.Error(codes.InvalidArgument, "Note must not be empty")
 	}
 	if n.Name == "" {
+		log.Printf("Note name must not be empty: %v", n.Name)
+		return nil, status.Error(codes.InvalidArgument, "Note name must not be empty")
+	}
+	pID, _, err := name.ParseNote(n.Name)
+	if err != nil {
 		log.Printf("Invalid note name: %v", n.Name)
 		return nil, status.Error(codes.InvalidArgument, "Invalid note name")
 	}
+	if _, err = g.Projects.GetProject(ctx, &pb.GetProjectRequest{Name: name.FormatProject(pID)}); err != nil {
+		log.Printf("Unable to get project %v, err: %v", pID, err)
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Project %v not found", pID))
```

Comment:
`codes.NotFound`

Reply:
Fixed in 55f8d67

Reply:
Should NotFound be returned even if the Project is missing? One could argue that NotFound should refer to the resource that was provided by the user (in this case the note). I think returning NotFound would make sense as well. Just want to double check.

Reply:
NotFound is what makes most sense to me. InvalidArgument should be reserved for things that never make sense - not for things that don't work with the current state of the world (eg that project doesn't exist, or the user doesn't have access).

Reply:
Agree. Fixed in 153e507.